doc/IDE_SETUP.md

```diff
@@ -0,0 +1,170 @@
+# Using IntelliJ IDEA and Cursive
+
+## Install Cursive
+
+See https://cursive-ide.com/userguide/index.html
+
+## Getting Cursive to understand `status-react`
+
+- Add this file to the root of the `status-react` project dir
+  - https://gist.github.com/Samyoul/f71a0593ba7a12d24dd0d5ef986ebbec
+- Right click and "add as leiningen project"
+
+![](https://notes.status.im/uploads/upload_32ea1a8eeb410effc24b5f3914e38627.png)
+
+## I get a lot of `cannot be resolved`
+
+Are you getting problems where you get a lot of `cannot be resolved` on everything?
+
+![](https://notes.status.im/uploads/upload_73ed33ce7af35a289c88afdb01fc91b7.jpg)
+
+See https://cursive-ide.com/userguide/macros.html
+
+- opt+enter (on macOS)
+- resolve defview as fn and letsubs as let
+- move selection on resolve and hit enter
+- and select defn for defview and let for letsubs
+
+## REPL!
+
+### Connecting to REPL to IntelliJ
+
+I had a number of problems connecting to REPL, the solution is as follows:
+
+At the top of IntelliJ IDEA click on the `Add Configuration...` option:
+
+![](https://notes.status.im/uploads/upload_b16dd48e03ef4cc98c53cc896a532f90.png)
+
+This will load the following menu:
+
+![](https://notes.status.im/uploads/upload_0f50285b745a00cfe3cab3c4af415d21.png)
+
+Click on the `+` icon in the top left corner of the menu.
+
+Select `Clojure REPL > Remote`
+
+![](https://notes.status.im/uploads/upload_0d890d5ed6e2d8d48d31cab7fa99d9e9.png)
+
+Which will load the following menu
+
+![](https://notes.status.im/uploads/upload_563dce472426f1f691a55713556a9b36.png)
+
+Enter the below options:
+
+- Name = status-react
+- Display Name = status-react
+- Connection type = nREPL
+- Connection details
+  - Host = 127.0.0.1
+  - Port = 7888
+
+![](https://notes.status.im/uploads/upload_88ed1d279c314d2264405f15a5e56433.png)
+
+Press `OK`
+
+Now the below option will be visible.
+Press the green run button
+
+![](https://notes.status.im/uploads/upload_d3c538110f5cc21770311b288ac382d7.png)
+
+You should now see an dialog with the following message:
+
+```shell
+Connecting to remote nREPL server...
+Clojure 1.10.1
+```
+
+To confirm you have a working connection with REPL enter the following command in the input box below the output:
+
+```shell
+(prn 1)
+```
+
+Which should output
+
+```shell
+(prn 1)
+1
+=> nil
+```
+
+Or
+
+```shell
+prn 1
+```
+
+Which should output
+
+```shell
+prn 1
+=> 1
+=> #function[clojure.core/prn]
+```
```

Comment:
This doesn't sound right `prn 1` is not valid without the parenthesis

Reply:
It works on my machine, you can see in the screenshot below. But it is a bit redundant so I've removed it from the docs